### ISSUE TITLE
null provisioner doesn't wait for resource to complete

### ISSUE THREAD

**Issue by scardena** [hashicorp/terraform#25680]
### Terraform Version

Terraform v0.12.28
+ provider.aws v2.70.0
+ provider.null v2.1.2
+ provider.template v2.1.2

### Terraform Configuration Files

```terraform
module "instance" {
  source = "../instance"
  rds_endpoint    = module.databases.rds_cluster_endpoint

}

module "databases" {
  source      = "../databases"
}
```

### Crash Output

```
 Exception message: (2003, \"Can't connect to MySQL server on 'my-db.cluster-a2dorobfadix.us-east-1.rds.amazonaws.com}' ([Errno -2] Name or service not known)\")"}
```

### Expected Behavior

Since instance module has a dependency on the databases module, instance module should wait for the databases module to finish. 

### Actual Behavior

Since the database takes too long to provision, terraform runs the null provisioners inside the instance module, before the database is fully provisioned, and fails because it cannot connect.

**Comment by alisdair** [comment]
I don't quite understand exactly what's going wrong here. Could you provide more of the configuration to help make it clearer?

Ideally, a minimal reproduction would make it easier for us to help. The simpler the better! Thanks in advance for your help.

**Comment by scardena** [comment]
I am closing this issue, because the problem was on my end. there were 2 mainly:
1. I had a typo on my connection string (extra `}` at the end)
2. I was waiting on the `module.databases.rds_cluster_endpoint` to be available, but I figured out I had to wait on `module.databases.rds_cluster_instance` resource, as the cluster can become available, but if there are no instance ready, a connection to the cluster won't work.
